### PR TITLE
NIFI-8606 Added Stop & Configure button to the Controller Services De…

### PR DESCRIPTION
**Pull request by Freedom9339**
…tails dialog

# Summary

[NIFI-8606](https://issues.apache.org/jira/browse/NIFI-8606) - Added a "Stop & Configure" button to the controller service details dialog. I added a status bar to the dialog showing the current status of the controller service, and the Stop & Configure button if the controller service is enabled. 

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [X] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by mosermw** [comment]
@Freedom9339  Your branch now has conflicts with two files.  If you could rebase onto main and resolve those conflicts, I will be happy to review.

**Comment by mosermw** [comment]
I tested this (after resolving conflicts locally) and it works on the main branch.  I will also test this on the support/nifi-1.x branch.  I like seeing the status-bar for controller service details, which makes it more consistent with the processor details display which has a status-bar.

Still need to spend some time in the code reviewing for best practices.  Maybe another committer with more NiFi UI experience can review for best practices also.

**Comment by mosermw** [comment]
I tested this on the support/nifi-1.x branch and this code works as expected there, too.

I would still like another javascript dev to review this.

**Comment by mcgilman** [comment]
Will review...

**Review by mcgilman — CHANGES_REQUESTED** [review]
@Freedom9339 Thanks for the PR!

@mosermw Thanks for the initial review!

Sorry for the delay reviewing this effort. I was happy to see that the work here accounts for concerns with referencing components of the service in question. In addition to the comment below, I noticed that the implemented controller services and referencing components are duplicated after the disable dialog is closed and the configuration dialog is shown. I'm attaching a screen capture of the issue.

disable-and-configure

Thanks again. Just a side note, can you please rebase your PR against current main rather than introducing a merge commit? The merge commit causes a messy git history making it a little more difficult to review the changes locally.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-controller-service.js [hunk 1]
```diff
@@ -2457,6 +2487,72 @@
...
+                    if (nfCommon.isDefinedAndNotNull(options.nfActions) ) {
+                        var cb = function() {
+                            $('#disable-controller-service-dialog').modal('hide');
+                            $("#controller-configuration-status-bar").statusbar('hideButtons');
+                            nfControllerService.showConfiguration(serviceTable, controllerServiceEntity);
+                        };
+
+                        var selection = nfCanvasUtils.getSelectionById(controllerServiceEntity.id);
+                        $("#controller-configuration-status-bar").statusbar('buttons',[{
+                            buttonHtml: '<i class="fa fa-stop stop-configure-icon" aria-hidden="true"></i><span>Stop & Configure</span>',
```
[the remark this answers is not in the document]
Comment (Freedom9339) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-controller-service.js]: Changed the button text to "Disable & Configure"

**Comment by Freedom9339** [comment]
> @Freedom9339 Thanks for the PR!
> 
> @mosermw Thanks for the initial review!
> 
> Sorry for the delay reviewing this effort. I was happy to see that the work here accounts for concerns with referencing components of the service in question. In addition to the comment below, I noticed that the implemented controller services and referencing components are duplicated after the disable dialog is closed and the configuration dialog is shown. I'm attaching a screen capture of the issue.
> 
> disable-and-configure [ disable-and-configure ](https://user-images.githubusercontent.com/123395/280101857-198c2e4c-5983-4229-83d6-f58aa445ac0b.gif) [ ](https://user-images.githubusercontent.com/123395/280101857-198c2e4c-5983-4229-83d6-f58aa445ac0b.gif)
> 
> Thanks again. Just a side note, can you please rebase your PR against current main rather than introducing a merge commit? The merge commit causes a messy git history making it a little more difficult to review the changes locally.

Fixed this. It should now only show each referenced component once.

**Comment by Freedom9339** [comment]
@mcgilman Thank you for the review. I've changed the text to show Disable instead of stopped, and fixed the issue where the referenced components were being shown twice.

**Comment by mosermw** [comment]
@mcgilman Are we at the point on the main branch where changes to nifi-web-ui also have to be made on the new nifi-web-frontend?  I'm not sure how they have been kept up-to-date.

**Comment by mosermw** [comment]
I tested this again on main and support/nifi-1.x branches.  The duplicated references is fixed.  So this looks good to me.

+1 will merge to main and 1.x

**Pull request closed without merging**